### DISCUSSION TITLE
Incremental Static Regeneration problem

### DISCUSSION THREAD

**Discussion by chriscoindeskindices** [vercel/next.js#47864] [Help]
### Summary

From the page:  https://nextjs.org/docs/basic-features/data-fetching/incremental-static-regeneration

> Note: The [experimental-edge runtime](https://nextjs.org/docs/api-reference/edge-runtime) is currently not compatible with ISR, although you can leverage stale-while-revalidate by setting the cache-control header manually.

I am using the revalidation field set to 60 seconds.  I am running on Vercel.  I did not opt in for experimental edge runtime (don't even know what it is), but I am not seeing updates.

```
export const getStaticProps: GetStaticProps<Props> = async () => {
  const dictionary: Dictionary = await getDictionary('Home');

  return {
    props: {
      dictionary,
      header: await getHeader(),
      footer: await getFooter()
    },
    revalidate: 60,
  };
};
```

Any help is appreciated. Thanks.

**Comment by icyJoseph** [comment]
Well, we have to guess now, but you could start by:

```ts
export const getStaticProps: GetStaticProps<Props> = async () => {
    try {
        const dictionary: Dictionary = await getDictionary('Home');

        return {
            props: {
                dictionary,
                header: await getHeader(),
                footer: await getFooter(),
            },
            revalidate: 60,
        };
    } catch (reason) {
        if (reason instanceof Error) {
            console.log(reason.message);
        } else {
            console.log('Unexpected Exception', reason);
        }

        throw reason;
    }
};
```

In this case, we re-throw to keep things from updating to a failed state, instead, since ISR fails, the page generated at build time is kept. Assuming of course the getStaticProps is running, and failing.

**Comment by chriscoindeskindices** [comment]
Thanks. I don't know what happened.  Didn't have a code change or anything except adding the try/catch.  No catch happened, but it's working now.

**Reply by icyJoseph** [reply]
Maybe you changed Node version on the deployment, or perhaps your CMS had issues, or it was rate limited. Maybe there was a disruption of service on Vercel's,... Going back to code without try/catch, works too right?
